### PR TITLE
Read files directly

### PR DESCRIPTION
**Pull request by onetonfoot**

### CODE REVIEW

**Comment by mcmcgrath13** [comment]
Thanks! Could you also update the type generation uses of JSON.read to no longer do the file checks? Seems like it shouldn't be necessary anymore with this change.

**Comment by mcmcgrath13** [comment]
Sorry for the add on, but could you also add a test for this?

**Comment by onetonfoot** [comment]
Good thing you asked for some test cases, it wasn't behaving as expected when reading types

**Comment by rafaqz** [comment]
Any chance we can get this merged?

File: src/gentypes.jl [hunk 1]
```diff
@@ -460,9 +456,7 @@ function generatetypes(
     root_name::Symbol = :Root,
 )
     # either a JSON.Array or JSON.Object
-    json = read_json_str(json_str)
-
-    # build a type for the JSON
+    json = read(json_str)
```
Comment (quinnj) [inline src/gentypes.jl:459]: wait, why are we not using `read_json_str` here anymore?

Comment (mcmcgrath13) [reply src/gentypes.jl:459]: `read` now does what `read_json_str` did, so it's redundant

Comment (quinnj) [reply src/gentypes.jl:459]: ah got it.

**Comment by quinnj** [comment]
@mcmcgrath13 does this look good to you? I think it LGTM

**Review by mcmcgrath13 — APPROVED** [review]
LGTM - Thank you!

**Pull request merged**